backend/src/ml_space_lambda/app_configuration/policy_helper/active_service_policy_manager.py

```diff
@@ -0,0 +1,187 @@
+                    {
+                        IAMStatementProperty.EFFECT: IAMEffect.DENY,
+                        IAMStatementProperty.ACTION: [
+                            "elasticmapreduce:*",
+                        ],
+                        IAMStatementProperty.RESOURCE: "*",
+                    },
+                    {
+                        IAMStatementProperty.EFFECT: IAMEffect.DENY,
+                        IAMStatementProperty.ACTION: ["iam:PassRole", "iam:ListRoleTags"],
+                        IAMStatementProperty.RESOURCE: [
+                            account_arn_from_context(context, "iam", f"role/{env_variables[EnvVariable.EMR_EC2_ROLE_NAME]}"),
+                            account_arn_from_context(
+                                context, "iam", f"role/{env_variables[EnvVariable.EMR_SERVICE_ROLE_NAME]}"
+                            ),
+                        ],
+                    },
+                    {
+                        IAMStatementProperty.EFFECT: IAMEffect.DENY,
+                        IAMStatementProperty.ACTION: ["ec2:AuthorizeSecurityGroupIngress"],
+                        IAMStatementProperty.RESOURCE: account_arn_from_context(context, "ec2", "security-group/*"),
+                    },
+                    {
+                        IAMStatementProperty.EFFECT: IAMEffect.DENY,
+                        IAMStatementProperty.ACTION: ["ec2:DescribeInstances", "ec2:DescribeRoutesTables"],
+                        IAMStatementProperty.RESOURCE: "*",
+                    },
+                ],
+                "ResourceType": ResourceType.EMR_CLUSTER,
+            },
+        }
+
+        # Instead of deleting the policy (which requires unattaching from all roles) assign a filler non-impactful statement
+        self.FILTER_DENY_STATEMENTS = [
+            {
+                IAMStatementProperty.EFFECT: IAMEffect.DENY,
+                IAMStatementProperty.ACTION: ["dynamodb:CreateTable"],
+                IAMStatementProperty.RESOURCE: account_arn_from_context(context, "dynamodb", "table/invalid-table"),
+            },
+        ]
+
+    def update_activated_services_policy(
+        self, app_configuration: AppConfigurationModel, iam_manager: IAMManager = None
+    ) -> list[ResourceType]:
+        """
+        Uses the provided app_configuration and either the provided iam_manager or a new instance of IAMManager to update the policies
+
+        Returns list[ResourceType] of resource types that should be suspended
+        """
+        resource_types_to_suspend = []
+        deny_policy_statements = []
+        enabled_services_dict = app_configuration.configuration.enabled_services.to_dict()
+
+        iam_manager = IAMManager() if iam_manager is None else iam_manager
+
+        # Check each service
+        for service in enabled_services_dict.keys():
+            # If the service is deactivated and has instructions on how to deny the service in 'service_disable_permissions'
```

Comment:
Does this comment need to be updated? `service_disable_permissions` only seems to appear in this comment